YubiKit/YubiKit/Helpers/Additions/YKFNSDataAdditions.m

```diff
@@ -248,7 +248,14 @@ - (BOOL)ykf_containsRange:(NSRange) range {
 @implementation NSData(NSData_Base32Additions)
 
 + (NSData *)ykf_dataWithBase32String:(NSString *)base32String {
-    return [self dataWithBase32String:base32String];
+    NSRange range = NSMakeRange(0, [base32String length]);
+    NSError *error = NULL;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern: @"[A-Za-z2-7=]*" options: 0 error: &error];
+    NSRange foundRange = [regex rangeOfFirstMatchInString: base32String options: 0 range: range];
```

Comment:
this seems to return your first match...which will be your first character, no?

Reply:
I think you should check the opposite...if range is not found then return nil
Otherwise you method might eturns nil for correct encoded string

Reply:
it returns a range of first continuous match